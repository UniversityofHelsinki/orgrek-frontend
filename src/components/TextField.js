import * as React from 'react';
import MuiTextField, {
  TextFieldProps as MuiTextFieldProps,
} from '@mui/material/TextField';
import Box from '@mui/material/Box';

const TextField = React.forwardRef((props, ref) => {
  const { onFocus, onBlur, helperText, ...other } = props;
  const maxLength = props.inputProps?.maxLength;

  const [visible, setVisible] = React.useState(false);

  return (
    <MuiTextField
      ref={ref}
      {...other}
      onFocus={(event) => {
        setVisible(true);
        onFocus && onFocus(event);
      }}
      onBlur={(event) => {
        setVisible(false);
        onBlur && onBlur(event);
      }}
      helperText={
        <Box
          component="span"
          sx={{ display: 'flex', justifyContent: 'space-between' }}
        >
          <span>{helperText}</span>
          {visible && maxLength > 0 && (
            <span>{`${props.value.length} / ${maxLength}`}</span>
          )}
        </Box>
      }
    />
  );
});

TextField.displayName = 'TextField';

export default TextField;
