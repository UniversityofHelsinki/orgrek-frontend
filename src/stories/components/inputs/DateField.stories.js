import React from 'react';
import TextField from '@mui/material/TextField';
import { DatePicker } from '@mui/x-date-pickers/DatePicker';
import { useState } from 'react';
import { parseISO } from 'date-fns';

export default {
  component: DatePicker,
  parameters: {
    docs: {
      description: {
        component: 'A wrapper component for MUI X date picker',
      },
    },
  },
};

export const Basic = {
  args: {
    label: 'Label',
    value: '',
    helperText: '',
    required: false,
    disabled: false,
    error: false,
  },
  render: (args) => {
    const [value, setValue] = useState(
      args.value ? parseISO(args.value) : null
    );

    return (
      <DatePicker
        {...args}
        value={value}
        onChange={(newValue) => setValue(newValue)}
        renderInput={(props) => (
          <TextField
            {...props}
            helperText={args.helperText}
            required={args.required}
            error={args.error}
          />
        )}
      />
    );
  },
};

export const FormProps = {
  ...Basic,
  args: {
    ...Basic.args,
    value: '2023-02-10',
    helperText: 'Assistive text',
    required: true,
  },
};

export const Validation = {
  ...Basic,
  args: {
    ...Basic.args,
    value: '2023-02-10',
    helperText: 'Validation error message',
    required: true,
    error: true,
  },
};
