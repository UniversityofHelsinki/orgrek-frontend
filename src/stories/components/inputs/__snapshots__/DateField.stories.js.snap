// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stories/components/inputs/DateField Basic smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Label",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Label",
      "pressed": undefined,
      "role": "textbox",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Choose date",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "stories-components-inputs-datefield--basic",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`stories/components/inputs/DateField FormProps smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Label",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "description": "Assistive text",
      "name": "Label",
      "pressed": undefined,
      "required": true,
      "role": "textbox",
      "value": "10.2.2023",
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Choose date, selected date is 10. helmi 2023",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Assistive text",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
  ],
  "name": "stories-components-inputs-datefield--form-props",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`stories/components/inputs/DateField Validation smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Label",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "description": "Validation error message",
      "invalid": "true",
      "name": "Label",
      "pressed": undefined,
      "required": true,
      "role": "textbox",
      "value": "10.2.2023",
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Choose date, selected date is 10. helmi 2023",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Validation error message",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
  ],
  "name": "stories-components-inputs-datefield--validation",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
